tcl/helpbrowser.tcl

```diff
@@ -268,10 +278,15 @@ proc ::helpbrowser::make_doclistbox {dir count {select true}} {
     check_destroy $count
     # exportselection 0 looks good, but selection gets easily out-of-sync
     set current_listbox [listbox "[set b .helpbrowser.frame.root$count]" \
-                             -yscrollcommand "$b-scroll set" \
-                             -highlightbackground white -highlightthickness 5 \
-                             -highlightcolor white -selectborderwidth 0 \
-                             -height 20 -width 24 -exportselection 0 -bd 0]
+        -yscrollcommand "$b-scroll set" -highlightbackground white \
+        -highlightthickness 5 -highlightcolor white -selectborderwidth 0 \
+        -height 20 -width 24 -exportselection 0 -bd 0 \
+        -background [::pdtk_canvas::get_color helpbrowser_fill .helpbrowser] \
+        -foreground [::pdtk_canvas::get_color helpbrowser_text .helpbrowser]]
+    set tmpcol [::pdtk_canvas::get_color helpbrowser_highlight .helpbrowser]
+    if {$tmpcol ne ""} {
+		$current_listbox configure -selectbackground $tmpcol
+	}
```

Comment:
why aren't the `-background` and `-foreground` properties made optional (like `-selectbackground`)?

Reply:
@umlaute I suppose I was again working under the assumption that we wanted consistent background and foreground color (black text on white background) for the helpbrowser

Reply:
@umlaeute I'm testing vanilla on dark theme osx mojave with tk 8.6.10. The default canvas is almost black! and I can barely see objects.. similarly, text in a text_define window is white on a white background (invisible)! it seems to me that this is an example of why the pdtk_canvas should draw white by default, and other colors should have a default too except for maybe for using the system highlight.. maybe this could be fixed through options in pd-gui if we really want the canvas subject to system themes but then we have to color many other things like object boxes according to system themes as well to make it work.